src/main/java/org/folio/services/piece/PieceService.java

```diff
@@ -87,13 +87,21 @@ private String buildUpdatePieceBatchQuery(Collection<Piece> pieces, String tenan
 
   public Future<Tx<PoLine>> updatePieces(Tx<PoLine> poLineTx, ReplaceInstanceRef replaceInstanceRef, DBClient client) {
     return getPiecesByPoLineId(poLineTx.getEntity().getId(), client)
-      .compose(pieces -> updateHoldingForPieces(poLineTx, pieces, replaceInstanceRef, client));
+      .compose(pieces -> updateHoldingForPieces(poLineTx, pieces, replaceInstanceRef, client))
+      .onComplete(ar -> {
+        if (ar.failed()) {
+          log.error("updatePieces(poLineTx, replaceInstanceRef, client) failed, poLineId={}",
+            poLineTx.getEntity().getId(), ar.cause());
+        } else {
+          log.debug("updatePieces(poLineTx, replaceInstanceRef, client) complete");
```

Comment:
should it be formatted string with specific values?

Reply:
Yes, I should add the po line id just in case there are other operations in parallel and the logs get confusing. I will add that.